### DISCUSSION TITLE
Renaming OrchardCore.Lucene to OrchardCore.Search.Lucene

### DISCUSSION THREAD

**Discussion by Skrypt** [OrchardCMS/OrchardCore#12198] [Polls]
As part of https://github.com/OrchardCMS/OrchardCore/pull/11052 we need to decide if we will rename the modules or not.

This means that we should also rename the abstraction modules too based on the naming convention that we started using with the "Filestorage" modules.

**Comment by hishamco** [comment]
Totally agree, it's align with what we did in `OC.Media`, `OC.Media.Azure` and `OC.Media.AmazonS3`

**Comment by micrub-it** [comment]
Hi,
I cat try to migrate a OC site version 1.3 to the current 1.5 implementation. When I access to any content by Admin or any deployment plan I have this error:

An unhandled exception occurred while processing the request.
JsonSerializationException: Could not load assembly 'OrchardCore.Lucene'.
Newtonsoft.Json.Serialization.DefaultSerializationBinder.GetTypeFromTypeNameKey(StructMultiKey<string, string> typeNameKey)
JsonSerializationException: Error resolving type specified in JSON 'OrchardCore.Lucene.Deployment.LuceneIndexDeploymentStep, OrchardCore.Lucene'. Path 'DeploymentSteps[7].$type', line 1, position 1533.


My site contains some deoplyment plans. One of this have this steps:

		{
			"$type": "OrchardCore.Lucene.Deployment.LuceneIndexDeploymentStep, OrchardCore.Lucene",
			"IncludeAll": true,
			"IndexNames": [],
			"Id": "6ee475a84ecb4e6da7062dcccfdf1c1c",
			"Name": "LuceneIndex"
		},
		{
			"$type": "OrchardCore.Lucene.Deployment.LuceneSettingsDeploymentStep, OrchardCore.Lucene",
			"Id": "84b99fcdf7af4700905a9f69e83d345c",
			"Name": "LuceneSettings"
		},

If I remove theese steps (manually from db, becose the Admin return the error) the problem are solved!

Why does an error related to deplyment plans occur when access to content by Admin?

And can you consider this possibility in your migration procedure? 
Thank you very much

**Reply by Skrypt** [reply]
Your deployment step "$type" should be now "OrchardCore.Search.Lucene.Deployment.LuceneIndexDeploymentStep, OrchardCore.Search.Lucene".

I will open an issue.

**Reply by Skrypt** [reply]
Issue is fixed here: https://github.com/OrchardCMS/OrchardCore/pull/12591

**Comment by micrub-it** [comment]
Thanks for your rapid fix,
I tested your fix with my deployment plan but I still have this exception:

exception has occurred while executing the request. Newtonsoft.Json.JsonSerializationException: Error resolving type specified in JSON 'OrchardCore.Lucene.Deployment.LuceneSettingsDeploymentStep, OrchardCore.Lucene'. Path 'DeploymentSteps[8].$type', line 1, position 1733.
 ---> Newtonsoft.Json.JsonSerializationException: Could not load assembly 'OrchardCore.Lucene'.
 (see attached full exception)

For my deployment plan, I think the error is this line is missing in the OrchardCore.Search.Lucene.Migrations:

updateCmd = $"UPDATE {dialect.QuoteForTableName(table)} SET Content = REPLACE(content, '\"$type\":\"OrchardCore.Lucene.Deployment.LuceneSettingsDeploymentStep, OrchardCore.Lucene\"', '\"$type\":\"OrchardCore.Search.Lucene.Deployment.LuceneSettingsDeploymentStep, OrchardCore.Search.Lucene\"') WHERE [Type] = 'OrchardCore.Deployment.DeploymentPlan, OrchardCore.Deployment.Abstractions'";

and so it worked!

[FullException.txt]
(https://github.com/OrchardCMS/OrchardCore/files/9741230/FullException.txt)

[deployment_plan_before_your_fix#12591.txt](https://github.com/OrchardCMS/OrchardCore/files/9741202/deployment_plan_before_your_fix.12591.txt)

[deployment_plan_after_your_fix#12591.txt](https://github.com/OrchardCMS/OrchardCore/files/9741204/deployment_plan_after_your_fix.12591.txt)

[deployment_plan_after_my_fix.txt]
(https://github.com/OrchardCMS/OrchardCore/files/9741205/deployment_plan_after_my_fix.txt)

**Reply by Skrypt** [reply]
Will be fixed by https://github.com/OrchardCMS/OrchardCore/pull/12594
I forgot to check if there were other deployment steps. I added the missing ones.

**Reply by PiemP** [reply]
Is it normal that I have to run the query without the slash before the double quote? Like this:

```
UPDATE OrchardCorePA.dbo.test__Document SET Content = REPLACE(content, '"$type":"OrchardCore.Lucene.Deployment.LuceneIndexDeploymentStep, OrchardCore.Lucene"', '"$type":"OrchardCore.Search.Lucene.Deployment.LuceneIndexDeploymentStep, OrchardCore.Search.Lucene"')       WHERE [Type] = 'OrchardCore.Deployment.DeploymentPlan, OrchardCore.Deployment.Abstractions';

UPDATE OrchardCorePA.dbo.test__Document SET Content = REPLACE(content, '"$type":"OrchardCore.Lucene.Deployment.LuceneSettingsDeploymentStep, OrchardCore.Lucene"', '"$type":"OrchardCore.Search.Lucene.Deployment.LuceneSettingsDeploymentStep, OrchardCore.Search.Lucene"')       WHERE [Type] = 'OrchardCore.Deployment.DeploymentPlan, OrchardCore.Deployment.Abstractions';

UPDATE OrchardCorePA.dbo.test__Document SET Content = REPLACE(content, '"$type":"OrchardCore.Lucene.Deployment.LuceneIndexResetDeploymentStep, OrchardCore.Lucene"', '"$type":"OrchardCore.Search.Lucene.Deployment.LuceneIndexResetDeploymentStep, OrchardCore.Search.Lucene"')       WHERE [Type] = 'OrchardCore.Deployment.DeploymentPlan, OrchardCore.Deployment.Abstractions';

UPDATE OrchardCorePA.dbo.test__Document SET Content = REPLACE(content, '"$type":"OrchardCore.Lucene.Deployment.LuceneIndexRebuildDeploymentStep, OrchardCore.Lucene"', '"$type":"OrchardCore.Search.Lucene.Deployment.LuceneIndexRebuildDeploymentStep, OrchardCore.Search.Lucene"')       WHERE [Type] = 'OrchardCore.Deployment.DeploymentPlan, OrchardCore.Deployment.Abstractions';

UPDATE OrchardCorePA.dbo.test__Document SET Content = REPLACE(content, '"$type":"OrchardCore.Lucene.LuceneQuery, OrchardCore.Lucene"', '"$type":"OrchardCore.Search.Lucene.LuceneQuery, OrchardCore.Search.Lucene"')       WHERE  [Type] = 'OrchardCore.Queries.Services.QueriesDocument, OrchardCore.Queries';

UPDATE OrchardCorePA.dbo.test__Document SET [Type] = 'OrchardCore.Search.Lucene.Model.LuceneIndexSettingsDocument, OrchardCore.Search.Lucene'       WHERE [Type] = 'OrchardCore.Lucene.Model.LuceneIndexSettingsDocument, OrchardCore.Lucene';
```

If I leave the slash before the double quote the REPLACE doesn't work.
